src/pages/New.js

```diff
@@ -38,7 +39,10 @@ export default function New() {
       <AddCircleIcon />
       <ModalTitle>Add Restaurant</ModalTitle>
       <ModalSection>Name</ModalSection>
-      <TextInput placeholder="Enter restaurant name" />
+      <TextInput
+        placeholder="Enter restaurant name"
+        onChange={event => setTitle(event.target.value)}
```

Comment:
good job! 
If you use an onChange it would make sense to set value too to control the component.
See https://reactjs.org/docs/forms.html#controlled-components